### DISCUSSION TITLE
Is there a way to open a file to existing neovide window?

### DISCUSSION THREAD

**Discussion by theammir** [neovide/neovide#1792] [Q&A] [answered]
I wanted to use Neovide as an external editor for Godot 4.0, but every time I double click a file, it opens a new window with it.
While I can use nvim explorer to navigate, it is a matter of searching for a file in both vim and godot every time I want to edit things.

Totally noob question, but Is there a workaround to just add a new tab or something if Neovide is running?

**Comment by alloc33** [comment]
Which OS do you use? Maybe it would help
https://github.com/neovide/neovide/issues/1324
https://github.com/neovide/neovide/issues/1259

**Comment by theammir** [comment, marked as answer]
https://github.com/neovide/neovide/issues/1324#issuecomment-1146836839
Spent a little more time than expected, but made this little powershell script, that somehow works better for my machine even though I barely made any changes.

```pwsh
$env:NVIM_LISTEN_ADDRESS = "\\.\pipe\nvim-nvr"
nvr.exe -s --nostart --servername "$env:NVIM_LISTEN_ADDRESS" -l $args; if ($LastExitCode -ne 0) { neovide -- --listen "$env:NVIM_LISTEN_ADDRESS" 
; if ($args) { nvr.exe -s --nostart --servername "$env:NVIM_LISTEN_ADDRESS" -l $args } } 
```

**Reply by Qwerty-Space** [reply]
How are you supposed to use this?

**Comment by ProHaller** [comment]
On Mac Os, I created a little automator application that allows me to do this: 

### Steps to Create an Automator Application

1. **Open Automator**: 
   - Go to your Applications folder and open Automator.

2. **Create New Application**:
   - Choose to create a new "Application" when prompted.

3. **Add Shell Script**:
   - Drag and drop the "Run Shell Script" action into the workflow area.
   - In the shell script action, set "Pass input" to "as arguments".
   - Write a shell script that opens files in Neovide. For example:

```bash
for f in "$@"
do
  /full/path/to/neovide "$f"
done
```
   
   This script loops through each file passed to the Automator application and opens them with Neovide.

4. **Save the Application**:
   - Save the Automator application with an appropriate name, like "OpenWithNeovide", to your Applications folder or another convenient location.

### Setting File Associations

1. **Right-click on a Code File**:
   - Find a code file in Finder, right-click on it, and select "Get Info".

2. **Change Open With**:
   - In the "Open with:" section of the info window, select your "OpenWithNeovide" application. If it's not listed, click "Other…" and navigate to where you saved your Automator application.

3. **Apply to All**:
   - Click "Change All…" to apply this setting to all files of the same type (e.g., all `.py` files if you did this for a Python file).

**Reply by proofer** [reply]
Excellent, thanks! However, it opens each file in a new instance of Neovide. Like the original poster, I'd like to open subsequent files in an existing instance, if there is one. Dragging a file from Finder into an open instance works, but it would be nice for double-clicking to work also.
